### ISSUE TITLE
Weird shadow outline around units on water

### ISSUE THREAD

**Issue by speed2CZ** [FAForever/fa#4696]
Noticed this with UEF units, including factory on a water.

![ForgedAlliance_2023-02-17_15-47-38](https://user-images.githubusercontent.com/10947024/219686768-274df86f-7967-4dcc-b7e2-6ad6362c86d9.jpg)

**Comment by speed2CZ** [comment]
Some wrecks on the surface
![ForgedAlliance_2023-02-17_16-30-58](https://user-images.githubusercontent.com/10947024/219696722-f0939bb2-f331-40a9-a2d2-6d48939078a0.jpg)

**Comment by Garanas** [comment]
This is not a shadow outline, but a problem with the units being refracted _in front_ of the water. We fixed it for some shaders, but didn't manage to fix it for others yet.

**Comment by BlackYps** [comment]
I made some observations that could help fix this bug. Interestingly the behaviour is different for different factions.
In SCFA everything seems to be working. With one small detail: Unit parts above the water get distorted as well and block the ground refraction from happening. It's not really noticeable as the water color is still applied. I think this is the best behaviour we can get out of this engine. I circled the area where it is happening, but you are only really able to see it in motion.
![Screenshot 2023-04-30 162936](https://user-images.githubusercontent.com/52536103/235358921-e36e0a64-2521-4067-af9d-913d8a4385dd.png)

In contrast, here is the current behaviour in faf:
![Screenshot 2023-04-30 160922](https://user-images.githubusercontent.com/52536103/235359106-fbcf8394-b440-4c8c-b6cf-472f4ec488e1.png)
Aeon is working mostly correctly, but it seems that where the refraction is missing the water color is missing as well. Interestingly Cybran and UEF behave as if the whole unit was under water and I am not sure what is happening with Seraphim. It looks like a combination of Aeon and UEF.
I hope this provides some clues what to look out for when trying to tackle this, but I have no good hypothesis why the factions behave in different ways.

The screenshots are made on Enfilade by the way, which is the perfect map to test this as the refraction scale is so large that it makes it way easier to see what is going on.

**Comment by Garanas** [comment]
@BlackYps it looks like this for me on that map:

![image](https://user-images.githubusercontent.com/15778155/235598104-477e6e98-f53a-440b-b3b5-7bb36a7c6d7e.png)

The 'orange shadows' are well visible here too. But the lighting is way different, do you have the last version?

**Comment by BlackYps** [comment]
I was using an earlier version of the map and the PBR shaders. When I use version 7 and fafdevelop then I get the same result as in your screenshot. So the units refracting over the sea level seems to be reintroduced by the PBR shader. I don't know why though, as it is working for Aeon and I couldn't find the cause of the difference between the factions.

**Comment by BlackYps** [comment]
I think the coloring of units with the water color is the culprit here. It needs to happen only for underwater parts, but apparently we don't get an undistorted mask for the parts of the unit above the water.

**Comment by speed2CZ** [comment]
I dont know about that one.

**Comment by BlackYps** [comment]
I did some investigation and found solutions. We can fix the cutout issue by simply coloring the water after we make these cutouts for the units. I'll make a PR in a moment.
Secondly, I found the reason why the PBR shaders are acting up. The mask that is used in the water shader that holds the unit parts above water uses the alpha channel of that unit's shader, which is controlling the glow of the unit. The original shaders all added a glowMinimum to the alpha channel. As I didn't know what good that was supposed to do, I skipped it for the PBR shaders (except Aeon for some reason). Adding the glowMinimum to all of them fixes the issue.
